### ISSUE TITLE
sls deploy deletes and re-creates APIG on every update

### ISSUE THREAD

**Issue by doapp-ryanp** [serverless/serverless#2530]
After initial creation via `sls deploy` I change something an then re-deploy.

I notice every time I run `sls deploy` after the initial time, the API Gateway Deployment is deleted then re-created.  After digging deeper, I see that the CloudFormation template that is generated for updates, puts a timestamp on the end of the CloudFormation resource name.  Because the timestamp is different on every update, CloudFormation treats it as needing to replace the resource.

Why does this need to be done?  I'm sure it is done for a good reason but I can't figure out why. 

This is problematic for a few reasons:
-  Un-needed churn (longer updates, slower development iterations)
-  No other resources/code can rely on the APIG deployment because the ARN changes
-  We don't allow programatic deletion of resources for security reasons so we are blocked

**Comment by flomotlik** [comment]
@doapp-jeremy it is you can read more about it in this issue: https://github.com/serverless/serverless/issues/1684

TL;DR: Without it any changes to API Gateway resources won't be redeployed as the name of the resource needs to change between deployments so a new APIG Deployment happens.
